specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -49,13 +48,298 @@ import type { WbMapping } from './mapping';
 import { parseWbMappings } from './mapping';
 import { fetchWbPickLists } from './pickLists';
 import { WbUploaded } from './Results';
-import { wbViewTemplate } from './Template';
-import { WbActions } from './WbActions';
-import { WbUtils } from './WbUtils';
+import { WbActionsComponent } from '../WbActions/WbActions';
+import { WbUtils, WbUtilsComponent } from './WbUtils';
 import { WbValidation } from './WbValidation';
+import { DataSetName } from './DataSetMeta';
+import { WbSpreadsheet } from './WbSpreadsheet';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { WbToolkit } from '../WbToolkit/WbToolkit';
 
 export type WbStatus = 'unupload' | 'upload' | 'validate';
 
+export type Workbench = {
+  dataset: Dataset;
+  cells: WbCellMeta;
+  disambiguation: Disambiguation;
+  validation: WbValidation;
+  data: RA<RA<string | null>>;
+  hot: Handsontable;
+  throttleRate: number;
+  mappings: WbMapping;
+  utils: WbUtils;
+  cellCounts: GetSet<WbCellCounts>;
+};
+
+export function WbViewReact({
+  dataset,
+  hotRef,
+  handleDatasetDelete,
+  triggerRefresh,
+  spreadsheetContainer,
+}: {
+  readonly dataset: Dataset;
+  readonly hotRef: React.RefObject<HotTable>;
+  readonly handleDatasetDelete: () => void;
```

Comment:
```suggestion
  onDatasetDeleted: handleDatasetDeleted,
  triggerRefresh,
  spreadsheetContainer,
}: {
  readonly dataset: Dataset;
  readonly hotRef: React.RefObject<HotTable>;
  readonly onDatasetDeleted: () => void;
```

(and do the same in the other components that have handleDatasetDelete prop)